### PR TITLE
print message if aws key and secret not available in test/data/creden…

### PR DESCRIPTION
**Pull request by Mo33n**
…tials for  process.env.AWS_PROFILE

*Issue #, if available:* [283](https://github.com/aws/aws-iot-device-sdk-js/issues/283)

*Description of changes:*
Print message if aws key and secret is not available in the credentials to file for AWS_PROFILE 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: device/index.js [hunk 1]
```diff
@@ -490,6 +490,13 @@ function DeviceClient(options) {
                }
                var user_profile = options.profile || process.env.AWS_PROFILE || 'default';
                var creds = getCredentials(fs.readFileSync(filename, 'utf-8'));
+
+                if (!creds[user_profile]) {
+                    console.log("WS Access Key ID and AWS Secret Key not found in file:" + options.filename + " for AWS_PROFILE:" + process.env.AWS_PROFILE);
+                    throw new Error(exceptions.INVALID_CONNECT_OPTIONS);
+                }
```
Comment (graebm) [inline device/index.js:497]: This explicit check isn't necessary, the code that loads the credentials file and gets data out is already wrapped in a try-catch that logs a decent message if something goes wrong:
`console.log('Failed to read credentials from ' + filename);`

then a few lines later, another useful message is logged and the INVALID_CONNECT_OPTIONS error is thrown:
```
// AWS Access Key ID and AWS Secret Key must be defined
if (isUndefined(awsAccessId) || (isUndefined(awsSecretKey))) {
   console.log('To connect via WebSocket/SigV4, AWS Access Key ID and AWS Secret Key must be passed either in options or as environment variables; see README.md');
   throw new Error(exceptions.INVALID_CONNECT_OPTIONS);
}
```

Maybe just improve the existing log message to include the user_profile it tried to load:
`console.log('Failed to read credentials for profile "' + user_profile + '" from ' + filename);`

You'll need to move the declaration of `var user_profile` outside the try-block, like `var filename`

**Review by graebm — CHANGES_REQUESTED** [review]

File: device/index.js (hunk 1 above)
Comment (Mo33n) [reply device/index.js]: Thanks, it was helpful.
I have updated the PR.

**Comment by Mo33n** [comment]
Hi, 
Its done.

File: device/index.js (hunk 1 above)
Comment (Mo33n) [reply device/index.js]: Up updated.

**Review by bretambrose — APPROVED** [review]

**Comment by bretambrose** [comment]
Merged in https://github.com/aws/aws-iot-device-sdk-js/pull/381

**Pull request closed without merging**